File: frontend/saijainstore/src/app/common/order.ts
import { OrderProduct } from "./order-product";
import { OrderState } from "./order-state";

export class Order {
    constructor(
        public id: number | null,
        public dateCreated: Date,
        public orderProducts: OrderProduct[],
        public userId: number,
        public state: OrderState
    ) { }

    getTotal(): number {
        let total = 0;
        this.orderProducts.forEach((orderProduct) => {
            total += orderProduct.price * orderProduct.quantity;
            console.log("total ", total)
        });
        return total;
    }
}

File: backend/src/main/java/com/gonzalodev/saiyajinstore/backend/domain/model/Order.java
package com.gonzalodev.saiyajinstore.backend.domain.model;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
public class Order {
    private Integer id;
    private LocalDateTime dateCreated;
    private List<OrderProduct> orderProducts;
    private OrderState state;
    private Integer userId;

    public Order() {
        orderProducts = new ArrayList<>();
    }

    public BigDecimal getTotalOrderPrice(){
        return orderProducts.stream().map( orderProduct -> orderProduct.getTotalItem()).reduce(BigDecimal.ZERO, BigDecimal::add);
    }
}
